mm2src/mm2_main/src/ext_api/one_inch/rpcs.rs

```diff
@@ -0,0 +1,437 @@
+use super::errors::ApiIntegrationRpcError;
+use super::types::{AggregationContractRequest, ClassicSwapCreateRequest, ClassicSwapLiquiditySourcesRequest,
+                   ClassicSwapLiquiditySourcesResponse, ClassicSwapQuoteRequest, ClassicSwapResponse,
+                   ClassicSwapTokensRequest, ClassicSwapTokensResponse};
+use coins::eth::{display_eth_address, wei_from_big_decimal, EthCoin, EthCoinType};
+use coins::{lp_coinfind_or_err, CoinWithDerivationMethod, MmCoin, MmCoinEnum};
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
+use trading_api::one_inch_api::client::ApiClient;
+use trading_api::one_inch_api::types::{ClassicSwapCreateParams, ClassicSwapQuoteParams, ProtocolsResponse,
+                                       TokensResponse};
+
+/// "1inch_v6_0_classic_swap_contract" rpc impl
+/// used to get contract address (for e.g. to approve funds)
+pub async fn one_inch_v6_0_classic_swap_contract_rpc(
+    _ctx: MmArc,
+    _req: AggregationContractRequest,
+) -> MmResult<String, ApiIntegrationRpcError> {
+    Ok(ApiClient::classic_swap_contract().to_owned())
+}
+
+/// "1inch_classic_swap_quote" rpc impl
+pub async fn one_inch_v6_0_classic_swap_quote_rpc(
+    ctx: MmArc,
+    req: ClassicSwapQuoteRequest,
+) -> MmResult<ClassicSwapResponse, ApiIntegrationRpcError> {
+    let (base, base_contract) = get_coin_for_one_inch(&ctx, &req.base).await?;
+    api_supports_coin(&base)?;
```

Comment:
Why is this check done for base coin only?

Reply:
fixed